core/corehttp/gateway_handler.go

```diff
@@ -785,6 +765,11 @@ func (i *gatewayHandler) buildIpfsRootsHeader(contentPath string, r *http.Reques
 	return rootCidList, nil
 }
 
+func webRequestError(w http.ResponseWriter, err error) {
+	re := err.(*requestError)
+	webError(w, re.Message, re.Err, re.StatusCode)
+}
```

Comment:
I like the idea to bundle unpacking here.
But if this function panics if it is not called with a `*requestError`.
This should really be in the signature instead (compiletime checks are better than the same check at runtime):
```suggestion
func webRequestError(w http.ResponseWriter, err *requestError) {
	webError(w, err.Message, err.Err, err.StatusCode)
}
```
I think all the `handle*` function you wrote would need to return `*requestError` too.

Reply:
I have pushed changes for your other feedback.  I'm not positive what you're wanting here though.  Are you suggesting that the functions should return both a `*requestError` and an `error`?  I need to read up on how panics are handled.

Reply:
I believe you're just saying I should explicitly use `*requestError` instead of `error`.  I've pushed those changes now.

Reply:
>  I believe you're just saying I should explicitly use *requestError instead of error.

Yeah exactly :)

I see how my `too` was confusing.
It was not "return `error` and `*requestError` too". But it was "do the same change as `webRequestError`" too.